bycycle/cyclepoints/extrema.py

```diff
@@ -89,7 +99,11 @@ def find_extrema(sig, fs, f_range, boundary=None, first_extrema='peak', filter_k
         last_decay = decay_xs[t_idx]
         next_rise = rise_xs[rise_xs > last_decay][0]
         # Identify time of trough
-        troughs[t_idx] = np.argmin(sig[last_decay:next_rise]) + last_decay
+        troughs[t_idx] = np.argmin(sig_pad[last_decay:next_rise]) + last_decay
+
+    # Remove padding
```

Comment:
I _think_ that instead of doing this down here, after the filter step, can't we just add:
```
if pad:
    sig, _ = remove_nans(sig)
```

If the padding is updated, as commented above, to pad both ends, the padding should line up exactly with the nan's added to the signal in `remove_filter_edges` (see: https://github.com/neurodsp-tools/neurodsp/blob/dec3b8167f7f0f402a8fc3bbfe14b87d94755d02/neurodsp/filt/utils.py#L211)

This way, the signal ends up padded, filtered, then cropped to expected around line 70, before it goes into the next procedures, and nothing needs to be fixed at the end.

Reply:
This ends up being a bit complicated. If the padding (two-sided) in `sig` and `sig_filt` are removed prior to the two `find_flank_zerox` calls, edge artifacts / missing extrema points will still occur. Here, the first and last peaks are missing when the padding/nans are removed prior the find_flank_zerox call. This is resolved by setting `remove_edges=False` when filtering, using the padded `sig` and `sig_filt` to compute peaks/troughs, and then removing/adjusting for padding on peaks/troughs.

Settings: `first_extrema=peak` and the signal starts with a peak and ends with a trough:

```
>           np.testing.assert_equal(peaks, maxima[0])
E           AssertionError: 
E           Arrays are not equal
E           
E           (shapes (98,), (100,) mismatch)
E            x: array([ 125,  225,  325,  425,  525,  625,  725,  825,  925, 1025, 1125,
E                  1225, 1325, 1425, 1525, 1625, 1725, 1825, 1925, 2025, 2125, 2225,
E                  2325, 2425, 2525, 2625, 2725, 2825, 2925, 3025, 3125, 3225, 3325,...
E            y: array([  25,  125,  225,  325,  425,  525,  625,  725,  825,  925, 1025,
E                  1125, 1225, 1325, 1425, 1525, 1625, 1725, 1825, 1925, 2025, 2125,
E                  2225, 2325, 2425, 2525, 2625, 2725, 2825, 2925, 3025, 3125, 3225,...
```

Here is the same signal / settings. The bottom plot shows the missing peaks when removing padding prior to `find_flank_zerox`.
![Screenshot_20201015_124508](https://user-images.githubusercontent.com/34786005/96178578-6ed95000-0ee4-11eb-9aef-ba93c272ec52.png)

I'm not really sure why this is happening, but it seems better to leave as is.